website/src/docs/hotchocolate/api-reference/migrate-from-11-to-12.md

```diff
@@ -45,3 +45,110 @@ services
     .AddGraphQLServer()
     .AddType(() => new UrlType("Url"));
 ```
+
+# Relay
+
+## Configuration
+
+Previously the configuration of the Relay integration was focused around the `EnableRelaySupport()` method. It allowed you to enable Global Object Identification and automatically adding a query field to mutation payloads.
+
+The problem is that `EnableRelaySupport()` always enables the Global Object Identification feature. This is not obviously implied by the name and also prevents you from using the other feature in isolation.
+
+Therefore we introduced two separate APIs to give you more explicit control over which parts of the Relay integration you want to enable.
+
+### Global Object Identification
+
+**OLD**
+
+```csharp
+services
+    .AddGraphQLServer()
+    .EnableRelaySupport();
+```
+
+**NEW**
+
+```csharp
+services
+    .AddGraphQLServer()
+    .AddGlobalObjectIdentification();
+```
+
+### Query field in Mutation payloads
+
+**OLD**
+
+```csharp
+services
+    .AddGraphQLServer()
+    .EnableRelaySupport(new RelayOptions
+    {
+        AddQueryFieldToMutationPayloads = true,
+        QueryFieldName = "rootQuery",
+        MutationPayloadPredicate = type => type.Name.Value.EndsWith("Result")
+    });
+```
+
+**NEW**
+
+```csharp
+sevices
+    .AddGraphQL()
+    .AddQueryFieldToMutationPayloads(options =>
+    {
+        options.QueryFieldName = "rootQuery";
+        options.MutationPayloadPredicate =
+            type => type.Name.Value.EndsWith("Result");
+    });
```

Comment:
I've always felt that adding query under a payload shouldn't be a Relay thing at all, so happy to see this separation as it's a feature that's useful to anyone; even if they don't follow the old Payload relay spec  (which has disappeared anyway).

Reply:
Yes, I've noticed its disappearance as well. But I still think it's valuable to keep around.